js/bundle/src/encoder.ts

```diff
@@ -60,22 +71,35 @@ export class BundleBuilder {
   }
 
   private addSection(name: string, content: CBORValue) {
-    if (this.sectionLengths.includes(name)) {
+    if (this.sectionLengths.find(s => s.name === name)) {
       throw new Error('Duplicated section: ' + name);
     }
-    this.sectionLengths.push(name);
-    this.sectionLengths.push(encodedLength(content));
+    let length: number;
+    if (name === 'responses') {
+      // The responses section can be large, so avoid using encodedLength().
```

Comment:
Oh, you are right. I meant `encodedLength(content)` shouldn't be used. Updated the comment.

The trick used here needs some knowledge about CBOR encoding to understand.

The `responses` section is an array, and CBOR's array encoding is the concatenation of:

1. variable-length integer encoding of the array length
2. CBOR-encoded contents of the array elements

`this.currentResponsesOffset` holds the length of 2, and the length of 1 can be calculated by `encodedLength(this.responses.length)`. The `encodedLength` operation is cheap heare, because the argument is just a number.

Reply:
Gotcha. Thanks for the clarification.